### PR TITLE
fix: check start date where also checking end date

### PR DESCRIPTION
**Pull request by axle-h**

### CODE REVIEW

**Review by marcus-bcl — APPROVED** [review]
Do we need tests for these conditions? Might be useful for if we switch to using `@Filter`.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/entity/OfficeLocation.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ import javax.persistence.Table
 
 @Entity
 @Table(name = "OFFICE_LOCATION")
-@Where(clause = "END_DATE IS NULL OR END_DATE > CURRENT_DATE")
+@Where(clause = "START_DATE <= CURRENT_DATE AND (END_DATE IS NULL OR END_DATE > CURRENT_DATE)")
```
Comment (electrified) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/entity/OfficeLocation.kt:14]: I think these checks are fine for new appointments, but historic appointments may be linked to office locations that are no longer current.

Do we not care as we don't have any get endpoints?

Comment (electrified) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/entity/OfficeLocation.kt:14]: I think the same consideration applies to staff and team.

Comment (electrified) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/entity/OfficeLocation.kt:14]: Or does the where not get applied once there is an associated entity linked by ID?

Comment (axle-h) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/entity/OfficeLocation.kt:14]: Hmm my understanding of this was that start & end dates act as soft deletion.
Delius checks them at least for team. I didnt confirm with office location. Ill have a look now.

**Comment by axle-h** [comment]
> Do we need tests for these conditions? Might be useful for if we switch to using `@Filter`.

yeah... we really need better seed data. Maybe I can search the existing seed for appropriate locations, teams & staff...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/entity/OfficeLocation.kt (hunk 1 above)
Comment (marcus-bcl) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/entity/OfficeLocation.kt:14]: Good point, in the Delius code it doesn't do the start date check for office location (from what I can tell).

**Pull request merged**